### DISCUSSION TITLE
Best CUDA settings ?

### DISCUSSION THREAD

**Discussion by Rudy34160** [vladmandic/sdnext#861] [Q&A]
Hi,
Could you tell me the settings of the CUDA tab that seem optimal to you for using this fork with an RTX 2060 12go?
I have already tried several different combinations, but I would like to be able to make the most of it.

**Comment by EfourC** [comment]
Anyone feel free to correct me, but probably the settings screenshotted below..

For the settings under `Stable Diffusion`, it's worth comparing `xformers` vs. `Scaled-Dot-Product` to see if there's a notable difference on your system, but they should be close.

If I remember correctly, you can check `SDP disable memory attention` (with Scaled-Dot-Product) to give deterministic results, at a slight performance drop (and memory usage increase? not sure, try it out).


<details> <summary>Click to expand...</summary

![chrome_aiDIqEckIB](https://github.com/vladmandic/automatic/assets/121468033/f7c81d2d-7ae7-45bf-b85e-1bc6c485909a)

![chrome_tGg0NqVwn0](https://github.com/vladmandic/automatic/assets/121468033/4825f4fa-8001-420d-afbb-544b8d104385)


</details>

**Comment by TomerGadol** [comment]
So for me with a 1660ti with the last commit I used "--no-half-VAE" and "enable upcast sampling" with Autocast and FP16.
Now after the latest update I need "--no-half" for model since upcast sampling doesn't work anymore.

**Reply by vladmandic** [reply]
upcast sampling is allowed **IF** card passes tests. if card fails tests, yes, that is ignored as it leads to many many issues.
and i gave one hint on what to try.

otherwise - i cannot count how many issues users reported with NaNs/black-images - I'd rather be on the safe-side than be overloaded with issues.
